src/tm_data_types/datum/waveforms/waveform.py

```diff
@@ -56,24 +113,155 @@ def operable_exclusive_metadata(self) -> Dict[str, Any]:
+        """
+        if self.extended_metadata is None:
+            self.extended_metadata = {}
+        self.extended_metadata.update(kwargs)
+
+    @classmethod
+    def remap(cls, lookup: dict[str, str], data: dict[str, Any], drop_non_existant: bool = False) -> dict[str, Any]:
+        """Remap the data to the correct format.
+        
+        This method transforms dictionary keys according to a lookup table, which is
+        commonly used when converting between different metadata formats or file types.
+        
+        Args:
+            lookup: Dictionary mapping source keys to target keys. Keys in 'data' that
+                   match keys in 'lookup' will be renamed to the corresponding values.
+            data: Dictionary of data to remap
+            drop_non_existant: If True, drop keys not found in lookup. If False, 
+                              preserve them unchanged (with a warning for unknown keys).
+            
         Returns:
-            A dictionary which provides the opposite naming convention.
+            Remapped dictionary with keys transformed according to lookup
+            
+        Example:
+            >>> lookup = {"old_name": "new_name", "another_old": "another_new"}
+            >>> data = {"old_name": "value1", "unknown_key": "value2", "another_old": "value3"}
+            >>> result = WaveformMetaInfo.remap(lookup, data, drop_non_existant=False)
+            >>> print(result)
+            {'new_name': 'value1', 'unknown_key': 'value2', 'another_new': 'value3'}
+            
+        Note:
+            - When drop_non_existant=False, unknown keys will trigger a UserWarning
+              suggesting the use of extended_metadata for custom fields
+            - This method is used internally by file format classes to convert
+              between different metadata schemas
         """
-        remapped_dict = {
-            lookup[key]: (val if not isinstance(val, Enum) else val.value)
-            for key, val in data.items()
-            if key in lookup or not drop_non_existant
-        }
+        remapped_dict = {}
+        unknown_keys = []
+        
+        for key, value in data.items():
+            if key in lookup:
+                remapped_dict[lookup[key]] = value
+            else:
+                if not drop_non_existant:
+                    remapped_dict[key] = value
+                    unknown_keys.append(key)
+        
+        # Provide helpful warning for unknown keys
+        if unknown_keys and not drop_non_existant:
+            import warnings
+            warnings.warn(
+                f"Unknown metadata fields {unknown_keys} will be preserved but may not be "
+                f"supported by all file formats. Consider using extended_metadata for custom fields.",
+                UserWarning,
+                stacklevel=2
+            )
+        
         return remapped_dict
 
+    extended_metadata: Optional[Dict[str, Any]] = None
```

Comment:
Can you move the dataclass attribute above the methods? That is a more standard organization for dataclasses.